/*
 * Copyright 2016 <Admobilize>
 * All rights reserved.
 */

syntax = "proto3";

package matrix_malos;

// configuration of drivers. Both global configuration and
// for specific drivers.
message DriverConfig {
  // Delay between updates. In seconds.
  float delay_between_updates = 1;
  // Timeout after last ping.
  float timeout_after_last_ping = 2;

  // Values for LEDs. When configuring the driver you should leave
  // delay_between_updates and timeout_after_last_ping unchanged.
  EverloopImage image = 3;
}

// Dummy lecture.
message Dummy {
  // Random value.
  float value = 1;
}

// Basic Humidity lecture.
message Humidity {
  // Humidity. 
  float humidity = 1;

  // Temperature.
  float temperature = 2;
}

// Basic Pressure lecture.
message Pressure {
  // Pressure.
  float pressure = 1;

  // Temperature.
  float temperature = 2;
}

// Basic Pressure lecture.
message Imu {
  //Vertical axis (yaw)
  float yaw = 1;

  //Lateral axis (pitch)
  float pitch = 2;

  //Longitudinal axis (roll)
  float roll = 3;
}

// Value for a led.
message LedValue {
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
  uint32 white = 4;
}

// The led array.
message EverloopImage {
  repeated LedValue led = 1;
}
